### PR TITLE
add webapp prepare initcontainer for security-proxy

### PR DESCRIPTION
**Pull request by jeanmi151**
I did it like that for "fast/easyway" to customize 404.jsp, 

Tested and working

### CODE REVIEW

**Comment by jeanpommier** [comment]
What's the purpose of this change ? I don't understand

**Comment by jeanmi151** [comment]
> What's the purpose of this change ? I don't understand

the purpose is to customize easly the 404.jsp page like here : https://dev.datagrandest.fr/gfds

File: templates/security-proxy/security-proxy-deployment.yaml [hunk 1]
```diff
@@ -50,6 +50,20 @@ spec:
...
           {{- end }}
       {{- end }}
+      - name: prepare-proxy-webapp
+        image: {{ $webapp.docker_image }}
+        command:
+          - /bin/sh
+          - -c
+          - cp -r /var/lib/jetty/webapps/ROOT/* /mnt/security_proxy_webapp ;
+            if [ -f /mnt/georchestra_datadir/security-proxy/ressources/ ] ; then
+              cp /mnt/georchestra_datadir/security-proxy/ressources/* /mnt/geoserver_webapp/ ;
```
Comment (jeanpommier) [inline templates/security-proxy/security-proxy-deployment.yaml:60]: What's geoserver_webapp stuff doing here ?

**Comment by jeanpommier** [comment]
Sorry, my mind is not in the SP right now. Can you explain a bit more in detail what it is doing and why ?
So far, what I understand: the idea is to 
  - copy the 404 HTML page from an external jetty container 
  - get the associated static resources from  the datadir /security-proxy/resources/. 
 
Why this split behaviour ? Why not putting it all in the datadir and making just a link (ln) at the right place for instance ?

File: templates/security-proxy/security-proxy-deployment.yaml (hunk 1 above)
Comment (jeanmi151) [reply templates/security-proxy/security-proxy-deployment.yaml:60]: a mistake haha

**Comment by jeanmi151** [comment]
> Sorry, my mind is not in the SP right now. Can you explain a bit more in detail what it is doing and why ? So far, what I understand: the idea is to
> 
>     * copy the 404 HTML page from an external jetty container
> 
>     * get the associated static resources from  the datadir /security-proxy/resources/.
> 
> 
> Why this split behaviour ? Why not putting it all in the datadir and making just a link (ln) at the right place for instance ?

we don't want all the webapp in the datadir, just little thing we want to customize
we are doing like for GN or GS here : https://github.com/georchestra/helm-georchestra/blob/main/templates/geoserver/geoserver-deployment.yaml#L49
1. copy the webapp ressources in the volume
2. overide few file we need to overide
3. mount this volume in the coutainer for it to run with custom webapp

**Comment by jeanmi151** [comment]
Why doing it like this ?
1. Is it done like that  for GeoNetwork, Geoserver and CAS
2. Don't want to go modify repo geOrchestra because process is long
3. Difficult to remake image with already version tag
4. If modify geOrchestra the good process would be to create new patch release

**Comment by jeanpommier** [comment]
I understand. I'm not against it per-se for a quick patch. But my fear is that it is going just to stack patch upon patch and clutter the helm chart with very specific stuff that shouldn't be there is the first place.
If we have to do it here, I'd really prefer adding the quite common logic of allowing to define in the values 
- extra initcontainers
- additional volumemounts to the containers 
which, I believe, would allow to do the same, but with more flexibility and genericity. And since this would be cluttering the values file, it's more visible and we might be more prone to port relevant improvements to upstream georchestra code

**Comment by jeanmi151** [comment]
@jeanpommier thanks for your comments, I think why want the same :)
In the mean time, Do you agree we merge this PR (I will test it is working) and we work on your propositions later with more time ? 
this ideas would be a great also : https://github.com/georchestra/georchestra/issues/4030 (we discussed about this with @edevosc2c )

**Comment by jeanpommier** [comment]
Yes, I'm OK with the temporary patch. I believe that with https://github.com/georchestra/georchestra/issues/4030 we are going in the right direction

**Comment by fvanderbiest** [comment]
I'm sorry, I'm really opposed to "temporary patches".
Often found in productions 10 years later...

Let's build something robust from the start :-)

**Comment by edevosc2c** [comment]
> Often found in productions 10 years later... 

That's not true for this one, we are going to propose a real solution over this temporary patch: https://github.com/georchestra/georchestra/issues/4030.

The temporary patch will be well gone once #4030 is implemented.

**Review by edevosc2c — APPROVED** [review]

File: templates/security-proxy/security-proxy-deployment.yaml [hunk 2]
```diff
@@ -50,6 +50,27 @@ spec:
...
           {{- end }}
       {{- end }}
+      - name: prepare-proxy-webapp
+        image: {{ $webapp.docker_image }}
+        command:
+          - /bin/sh
+          - -c
+          - cp -r /var/lib/jetty/webapps/ROOT/* /mnt/security_proxy_webapp ;
+            if [ -d /mnt/georchestra_datadir/security-proxy/ressources/ ] ; then
+              cp /mnt/georchestra_datadir/security-proxy/ressources/* /mnt/security_proxy_webapp/ ;
```
Comment (pmauduit) [inline templates/security-proxy/security-proxy-deployment.yaml:60]: --force by default ?

Comment (jeanmi151) [reply templates/security-proxy/security-proxy-deployment.yaml:60]: without it seems working, as here : https://github.com/georchestra/helm-georchestra/pull/52/files#diff-b058540d36c74466303361ec6dd03c7d0adf765a2bdc339a46dde277182ec376R51

Comment (pmauduit) [reply templates/security-proxy/security-proxy-deployment.yaml:60]: looks like, the other cp being made on other places don't seem to require the option.

**Pull request merged**